addons/binding/org.openhab.binding.innogysmarthome/src/main/java/org/openhab/binding/innogysmarthome/handler/InnogyBridgeHandler.java

```diff
@@ -0,0 +1,843 @@
+            return true;
+
+            // Controller offline
+        } else if (e instanceof ControllerOfflineException) {
+            logger.error("innogy SmartHome Controller is offline.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, e.getMessage());
+            dispose();
+            scheduleReinitialize();
+            return false;
+
+            // Configuration error
+        } else if (e instanceof ConfigurationException) {
+            logger.error("Configuration error: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            dispose();
+            return false;
+
+            // invalid auth code
+        } else if (e instanceof InvalidAuthCodeException) {
+            logger.error("Error fetching access tokens. Invalid authcode! Please generate a new one. Detail: {}",
+                    e.getMessage());
+            org.eclipse.smarthome.config.core.Configuration configuration = editConfiguration();
+            configuration.put(CONFIG_AUTH_CODE, "");
+            updateConfiguration(configuration);
+            config.setAuthCode("");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Invalid authcode. Please generate a new one!");
+            dispose();
+            return false;
+
+        } else if (e instanceof InvalidActionTriggeredException) {
+            logger.error("Error triggering action: {}", e.getMessage());
+            return true;
+
+            // io error
+        } else if (e instanceof IOException) {
+            logger.error("IO error: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            dispose();
+            scheduleReinitialize(REINITIALIZE_DELAY_LONG_SECONDS);
+            return false;
+
+            // unexpected API error
+        } else if (e instanceof ApiException) {
+            logger.error("Unexcepted API error: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            dispose();
+            scheduleReinitialize(REINITIALIZE_DELAY_LONG_SECONDS);
+            return false;
+
+            // java.net.SocketTimeoutException
+        } else if (e instanceof SocketTimeoutException) {
+            logger.error("Socket timeout: {}", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            dispose();
+            scheduleReinitialize();
+            return false;
+
+            // ExecutionException
+        } else if (e instanceof ExecutionException) {
+            logger.error("ExecutionException: {}", ExceptionUtils.getRootCauseMessage(e));
+            dispose();
+            scheduleReinitialize();
+            return false;
+
+            // unknown
+        } else {
```

Comment:
You mean `handleClientException` should simply throw an IllegalStateException up? In my experience with the binding it is more helpful to act and restart the binding as this exception should normally not occur. I changed the logging to `debug`.